tests/data/hrrr/test_file_loader.py

```diff
@@ -0,0 +1,177 @@
+import unittest
+
+import mock
+import xarray
+
+from tests.RME_test_case import RMETestCase
+from weather_forecast_retrieval.data.hrrr.file_loader import FileLoader
+from weather_forecast_retrieval.data.hrrr.grib_file import GribFile
+from weather_forecast_retrieval.data.hrrr.netcdf_file import NetCdfFile
+
+
+class TestFileLoader(unittest.TestCase):
+    def setUp(self):
+        self.subject = FileLoader()
+
+    def test_file_loader_property(self):
+        self.assertEqual(None, self.subject.file_loader)
+
+        self.subject.file_loader = GribFile
+
+        self.assertEqual(GribFile, self.subject.file_loader)
+
+    def test_file_type(self):
+        self.assertEqual(None, self.subject.file_type)
+
+        self.subject.file_loader = GribFile
+
+        self.assertEqual(GribFile.SUFFIX, self.subject.file_type)
+
+
+def saved_data_return_values():
+    metadata = mock.MagicMock()
+    metadata.name = 'metadata'
+    dataframe = mock.MagicMock(spec={})
+    dataframe.name = 'dataframe'
+    return metadata, dataframe
+
+
+@mock.patch.object(FileLoader, 'get_data')
+@mock.patch.object(
```

Comment:
I only slightly understand mock. So the patch here returns 2 values, the `metadata` and `dataframe` object when `get_data` is called? This mock objects are then what's passed to the test as `_data_patch` and `_df_patch`?

Reply:
The way this mock setup works is this:
* use the `@mock` decorators to mock the calls to the specified methods. They are only used, however, if you also pass them to each `test_` method. Difference between the `_` and non-underscore version is simply to indicate unused variables (flake8).
* The method `saved_data_return_values` then lets you specify what gets returned as it was set as an parameter for the second mock.